### ISSUE TITLE
Update version pinning logic to insert exact version matches

### ISSUE THREAD

**Issue by mscottford** [corgibytes/freshli-agent-dotnet#22]
Per the [Nuget documentation for versions](https://learn.microsoft.com/en-us/nuget/concepts/package-versioning), a version expression in the form of `1.0` is _actually_ specifying that any version `>= 1.0` is valid. That means that we need to update our version pinning logic to treat all version expressions in the form of `1.0` to be `[1.0, latest]` which would pin the upper bound of the resolution to be the latest version that was published before (or on) the `asOf` date/time.

**Comment by mscottford** [comment]
I think I misunderstood the documentation when I read it the first time. I find the table that talks about version ranges to be a bit confusing when compared to the annotated examples that appear later in the document.

Specifically this one:
```
<!-- Accepts any version 6.1 and above.
     Will resolve to the smallest acceptable stable version.-->
<PackageReference Include="ExamplePackage" Version="6.1" />
```

I read that as saying that if version `6.1` gets yanked and `6.2` is available, then `6.2` will be used instead of throwing an error message.

So I'm not sure the best way that we should pin these packages, then... the goal of the pinning process is to prevent the package resolution from resulting in a package version that was released after the `asOf` date. I'm trying to figure out the best way to do that and still preserve the semantics.

**Comment by mscottford** [comment]
After investigated further, I don't think this is actually an issue.
